### DISCUSSION TITLE
URL query parameters work fine in DEV but not once deployed on Vercel

### DISCUSSION THREAD

**Discussion by sherrikhan** [vercel/next.js#62855] [Help]
### Summary

Next.JS 14 app 

In my app's page.tsx I'm looking for a value passed on the URL.

```
const Home = ({
  searchParams,
}: {
  searchParams: { [key: string]: string | string[] | undefined };
}) => {

console.log("URL query parameters: ", searchParams);
```

In DEV on localhost, I can pass it in value and the app sees it. Once deployed on vercel, my app doesn't seem to be able to read any query parameters.

Is there perhaps a security setting that I need to turn off?

**Comment by icyJoseph** [comment]
Hi,

This is strange. Do you see the same issue if you build the application locally? Not in DEV but production mode, as in `next build && next start`.


Do you maybe have  a `export const dynamic` declaration?

**Reply by sachinkaplan** [reply]
@icyJoseph 
I am too facing same issue.

Tried `next build && next start` - works fine on local
Also tried adding await as params and searchParams are Promise but didn't help

**Reply by icyJoseph** [reply]
Do you have a link to a repository? Or maybe you have a vercel.json file? Super odd issue...

**Reply by sachinkaplan** [reply]
@icyJoseph Thanks for your reply.

Did some digging and found out its happening for few of our domains and not all. With this looks like issue occurred due to server URL redirects and not the NextJS code.

**Reply by icyJoseph** [reply]
Aha, yah that would, somewhat explain the issue. Do you have a solution?

**Reply by AmicoderDev** [reply]
Facing the same issue. I am not sure why it's happening.
